### PR TITLE
feat: selections in virtualized table

### PR DESCRIPTION
**Pull request by cbt1**
Adds support for single selections and multi selections by dragging.

The implementation is using some logic from the ordinary table, but I was not able to use it all, without modifying the shared code. Which would have added a risk to the ordinary table. So I opted for a solution where I used the logic that I could.

### CODE REVIEW

File: src/table/components/virtualized-table/__tests__/TableContainer.spec.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import React from 'react';
 import { render, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
```
Comment (cbt1) [inline src/table/components/virtualized-table/__tests__/TableContainer.spec.tsx:3]: Had to use `@testing-library/user-event` to get the proper chain of events triggerd by a click

File: src/table/components/virtualized-table/__tests__/TableContainer.spec.tsx [hunk 2]
```diff
@@ -69,7 +106,90 @@ describe('<TableContainer />', () => {
...
+
+      const { getByText, user } = renderTableContainer();
+
+      await waitFor(() => expect(getByText(cellText)).toBeVisible());
+
+      await waitFor(() => expect(getByText(cellText).parentNode).not.toHaveClass('selected'));
+
+      user.click(getByText(cellText));
+
+      await waitFor(() => expect(getByText(cellText).parentNode).toHaveClass('selected'));
```
Comment (cbt1) [inline src/table/components/virtualized-table/__tests__/TableContainer.spec.tsx:128]: Don't really like the idea of checking a class exists or not. But only other option I could come up with was to add a testid which would add some additional logic just to make the code slight better in terms of testability. Still undecided on this I guess.

File: src/table/components/virtualized-table/hooks/use-selector.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import { SelectionStates } from '../../../constants';
+import { TableContext, useContextSelector } from '../../../context';
+import { CellStyle } from '../../../types';
+import { getSelectionMouseHandlers } from '../../../utils/handle-click';
+import { getCellSelectionState } from '../../../utils/selections-utils';
+import { getSelectionStyle } from '../../../utils/styling-utils';
+
+const NOOP_ANNOUNCE = () => {};
+
+const PREVENT_DEFAULT = (e: React.MouseEvent<HTMLTableCellElement>) => e.preventDefault();
```
Comment (haxxmaxx) [inline src/table/components/virtualized-table/hooks/use-selector.ts:11]: is it common practice to do these in snake case?

File: src/table/components/virtualized-table/Cell.tsx [hunk 4]
```diff
@@ -19,32 +21,38 @@ const Cell = ({ columnIndex, rowIndex, style, data }: CellProps) => {
...
     return (
       <div
         style={{
           ...style,
+          ...bodyStyle,
+          ...selectionStyling,
           display: 'flex',
           alignItems: 'center',
-          borderColor: bodyStyle.borderColor,
-          borderStyle: 'solid',
```
Comment (haxxmaxx) [inline src/table/components/virtualized-table/Cell.tsx:30]: I have remove borderStyle from the styling-utils. There are no borders now when I am testing it

**Review by haxxmaxx — APPROVED** [review]
Looks good in general. How much effort do you think it would be to align the two implementations? what parts would you not need to add here in that case? Reasonable to do it later?

File: src/table/components/virtualized-table/Cell.tsx (hunk 4 above)
Comment (cbt1) [reply src/table/components/virtualized-table/Cell.tsx:31]: Fixed that now

File: src/table/components/virtualized-table/hooks/use-selector.ts (hunk 3 above)
Comment (cbt1) [reply src/table/components/virtualized-table/hooks/use-selector.ts:11]: Actually no, will fix

**Pull request merged**